tests/components/TopQueries/TopQueries.spec.ts

```diff
@@ -0,0 +1,176 @@
+import { IQuerySuggestResponse, NoopAnalyticsClient } from 'coveo-search-ui';
+import { Mock } from 'coveo-search-ui-tests';
+import { createSandbox, SinonSandbox } from 'sinon';
+import { ITopQueriesOptions } from '../../../src/components/TopQueries/TopQueries';
+import { TopQueries } from '../../../src/Index';
+
+describe('TopQueries', () => {
+    let sandbox: SinonSandbox;
+    const options: ITopQueriesOptions = {
+        suggestionQueryParams: {
+            q: '',
+            searchHub: 'test',
+        },
+    };
+
+    const EMPTY_SUGGESTION: IQuerySuggestResponse = {
+        completions: [],
+    };
+
+    const SUGGESTION: IQuerySuggestResponse = {
+        completions: [
+            {
+                expression: 'test1',
+                score: 1,
+                highlighted: 'testHighlight',
+                executableConfidence: 1,
+            },
+        ],
+    };
+
+    const SUGGESTIONS: IQuerySuggestResponse = {
+        completions: [
+            {
+                expression: 'test1',
+                score: 1,
+                highlighted: 'testHighlight',
+                executableConfidence: 1,
+            },
+            {
+                expression: 'test2',
+                score: 1,
+                highlighted: 'testHighlight',
+                executableConfidence: 1,
+            },
+            {
+                expression: 'test3',
+                score: 1,
+                highlighted: 'testHighlight',
+                executableConfidence: 1,
+            },
+        ],
+    };
+
+    beforeAll(() => {
+        sandbox = createSandbox();
+    });
+
+    afterEach(() => {
+        sandbox.restore();
+    });
+
+    it('should hide itself if there are no suggestions', async () => {
+        let topQueries = Mock.basicComponentSetup<TopQueries>(TopQueries, options);
+        let stub = sandbox.stub(topQueries.env.searchEndpoint, 'getQuerySuggest');
```

Comment:
You copy-🍝 the setup phase of your tests a lot, you may want to move them to a beforeEach close or a function.
The balance between Do Repeat Yourself in test and the classic Don't Repeat Yourself is finicky, so that could be considering an opinion (i.e. feel free to ignore me on this one ;P )

Reply:
Thanks a lot for all the great feedback!😄